make label_image.py v2 compatible

freedomtan opened the pull request:
Update `tensorflow/examples/label_image/label_image.py` to make it V2 compatible.

freedomtan committed:
make label_image.py v2 compatible

gbaned commented:
@freedomtan Could you please resolve the conflicts? Thanks!

freedomtan committed:
Merge branch 'master' into label_image_py_v2

caisq commented on tensorflow/examples/label_image/label_image.py:
```diff
@@ -40,33 +40,37 @@ def read_tensor_from_image_file(file_name,
-  result = sess.run(normalized)
+    float_caster = tf.cast(image_reader, tf.float32)
+    dims_expander = tf.expand_dims(float_caster, 0)
+    resized = tf.image.resize(dims_expander, [input_height, input_width], method=tf.image.ResizeMethod.BILINEAR)
```
style nit: Respect the 80-char line width limit.

freedomtan replied:
oops. thanks

freedomtan committed:
wrap line for 80-char line width

gbaned commented:
@freedomtan Could you please check failed build errors? Thanks!

freedomtan commented:
@gbaned it seems none of them are relevant. Maybe the master branch happened to have problems. E.g., in the [Ubuntu Python3 PIP](https://source.cloud.google.com/results/invocations/1f017dbe-c933-463f-a739-c6fd1291ae9d/log) log, the problem is in LLVM related code.

gbaned commented:
@freedomtan Thank you. Can you please check Ubuntu Sanity failures?

freedomtan commented:
I don't know how to check it. I can see logs of Ubuntu Python3 PIP, Ubuntu contrib, and Windows Bazel GPU. But not Ubuntu Sanity and Windows Bazel. See the figure below

![image](https://user-images.githubusercontent.com/3395998/62347796-7b66c800-b52d-11e9-8c14-e13b96296e4c.png)

gbaned commented:
@freedomtan Sorry for the late response. Can you please check Ubuntu Sanity failures now? Thank you.

freedomtan committed:
make pylint happy

1. add spaces for wrong hanging indentation reported by CI
2. add docstrings
3. remove 2 unused varibles

freedomtan commented:
@gbaned Thanks. Yes, I saw them. Fixed pylint Cs.

mellanox-github commented:
Can one of the admins verify this patch?

gbaned commented:
@caisq Can you please take a look on this PR? Thanks!

caisq commented:
Adding @MarkDaoust to reviewers. He's looking into the examples directory recently.

MarkDaoust commented:
@freedomtan, thanks for taking the time to put together this PR.

But even with these updates, this is not the sort of code we want to encourage:

    1. Nobody should be using raw GraphDef's anymore.
    2. I'd rather not have any examples directly using Session.

How did you get pointed to this example? someone should update that link.

Modern examples of how to do something like this is included in: 

https://github.com/tensorflow/docs/blob/master/site/en/tutorials/images/transfer_learning_with_hub.ipynb
https://github.com/tensorflow/docs/blob/master/site/en/guide/saved_model.ipynb

I'm working on deleting this examples/ directory. Please submit future PRs to https://github.com/tensorflow/examples

freedomtan commented:
@MarkDaoust I wrote it originally :)

MarkDaoust commented:
> I wrote it originally :)

😲

I do appreciate you taking the time to come back and update it.
And I hope you understand that I mean no disrespect to you or your work.

nanshihui commented:
i use  label_image.py to predict a lot of picture .however the memory leak. anyone find it ?

MarkDaoust commented:
Hi @nanshihui please see the examples I linked to previously for a modern way to do this.

nanshihui commented:
i turn hub.Module params    trainable to True,
it return 
```
tensorflow.python.framework.errors_impl.InvalidArgumentError: You must feed a value for Placeholder  with dtype float and shape [?,299,299,3]

```
do you have any ways? @MarkDaoust
